play-routes-compiler/src/main/scala/CommandLinePlayRoutesCompiler.scala

```diff
@@ -3,74 +3,93 @@ package rulesplayroutes.routes
 import com.google.devtools.build.buildjar.jarhelper.JarCreator
 import higherkindness.rules_scala.common.error.AnnexWorkerError
 import higherkindness.rules_scala.common.worker.WorkerMain
+import higherkindness.rules_scala.common.sandbox.SandboxUtil
 import java.io.{File, PrintStream}
-import java.nio.file.{Files, Paths}
+import java.nio.file.{Files, Path, Paths}
 import play.routes.compiler._
 import play.routes.compiler.RoutesCompiler.RoutesCompilerTask
+import scala.collection.mutable.ListBuffer
 import scala.jdk.CollectionConverters._
 import scala.io.Source
+import scala.util.{Try, Success, Failure}
+import scopt.Read
 import scala.Console._
 import scopt.OptionParser
 
 object CommandLinePlayRoutesCompiler extends WorkerMain[Unit] {
+  implicit val listPathRead: Read[List[Path]] = Read.reads { commaList =>
+    commaList.split(',').view.map(Paths.get(_)).toList
+  }
 
   case class Config(
-    sources: Seq[File] = Seq.empty[File],
-    generatedDirectory: File = new File("."),
-    outputSrcJar: File = new File("."),
-    additionalImports: Seq[String] = Seq.empty[String],
-    routesGenerator: RoutesGenerator = InjectedRoutesGenerator,
-    generateReverseRouter: Boolean = false,
-    namespaceReverserRouter: Boolean = false,
-    generateForwardsRouter: Boolean = true
+    var sources: List[Path] = List.empty[Path],
+    var outputDirectory: Path = Paths.get("."),
+    var outputSrcJar: Path = Paths.get("."),
+    var additionalImports: List[String] = List.empty[String],
+    var routesGenerator: RoutesGenerator = InjectedRoutesGenerator,
+    var generateReverseRouter: Boolean = false,
+    var namespaceReverserRouter: Boolean = false,
+    var generateForwardsRouter: Boolean = true
   )
 
-  val parser = new OptionParser[Config]("scopt") {
+  def parser(workDir: Path) = new OptionParser[Config]("scopt") {
     head("Command Line Play Routes Compiler", "0.1")
 
-    arg[File]("<outputDirectory>").required().action { (value, config) =>
-      config.copy(generatedDirectory = value)
+    arg[Path]("<outputDirectory>").required().action { (outputDirectory, config) =>
+      config.outputDirectory = SandboxUtil.getSandboxPath(workDir, outputDirectory)
```

Comment:
Nit: I'm not a huge fan of the mutability here. What's wrong with `.copy`? If we're trying to micro-optimize, are the benefits really worth it?

Reply:
Speed is the reason. I do think it's worth it because that many object copies is not very fast.

Also I'm not too worried about it as the only uses of this object are in this file, it's very short, and reasoning about correctness is much easier than a larger codebase.